src/backend/CassandraBackend.h

```diff
@@ -72,18 +72,28 @@ class BasicCassandraBackend : public BackendInterface
         if (auto const res = handle_.connect(); not res)
             throw std::runtime_error("Could not connect to Cassandra: " + res.error());
 
-        if (auto const res = handle_.execute(schema_.createKeyspace); not res)
+        if (!readOnly)
         {
-            // on datastax, creation of keyspaces can be configured to only be done thru the admin interface.
-            // this does not mean that the keyspace does not already exist tho.
-            if (res.error().code() != CASS_ERROR_SERVER_UNAUTHORIZED)
-                throw std::runtime_error("Could not create keyspace: " + res.error());
-        }
-
-        if (auto const res = handle_.executeEach(schema_.createSchema); not res)
-            throw std::runtime_error("Could not create schema: " + res.error());
+            if (auto const res = handle_.execute(schema_.createKeyspace); not res)
+            {
+                // on datastax, creation of keyspaces can be configured to only be done thru the admin interface.
+                // this does not mean that the keyspace does not already exist tho.
+                if (res.error().code() != CASS_ERROR_SERVER_UNAUTHORIZED)
+                    throw std::runtime_error("Could not create keyspace: " + res.error());
+            }
 
-        schema_.prepareStatements(handle_);
+            if (auto const res = handle_.executeEach(schema_.createSchema); not res)
+                throw std::runtime_error("Could not create schema: " + res.error());
+        }
+        try
+        {
+            schema_.prepareStatements(handle_);
+        }
+        catch (std::exception const& ex)
```

Comment:
Is there maybe a finer exception type we can catch here? What if some unrelated exception happens and readOnly was not actually set to true?

Reply:
I change the type to runtime_error which is raised here -> https://github.com/XRPLF/clio/blob/a76194d299f2b853ba865447a5f45f17a9ebac56/src/backend/cassandra/Handle.cpp#L152. 
I think we may have other exception happens but readonly is not true. So I print out the readonly value.

Reply:
It's good you print it - i'm only concerned with the formatting of the error message.. users will have a hard time if it's all collapsed and have no spaces.